common/persistence/cassandra/mutable_state_store.go

```diff
@@ -679,6 +682,10 @@ func (d *MutableStateStore) UpdateWorkflowExecution(
 		_ = conflictIter.Close()
 	}()
 
+	var dbVersion int64 = 0
+	if updateWorkflow.DBRecordVersion-1 > dbVersion {
```

Comment:
Instead of continue supporting next event ID as CAS criteria, why not removing legacy execution tests?

Reply:
If we decide to have this change, I will make the change